fedimint-core/src/encoding/bls12_381.rs

```diff
@@ -0,0 +1,88 @@
+use bls12_381::{G1Affine, G2Affine, Scalar};
+
+use crate::encoding::{Decodable, Encodable};
+use crate::module::registry::ModuleDecoderRegistry;
+
+impl Encodable for Scalar {
+    fn consensus_encode<W: std::io::Write>(&self, writer: &mut W) -> Result<usize, std::io::Error> {
+        self.to_bytes().consensus_encode(writer)
+    }
+}
+
+impl Decodable for Scalar {
+    fn consensus_decode<D: std::io::Read>(
+        d: &mut D,
+        _modules: &ModuleDecoderRegistry,
+    ) -> Result<Self, crate::encoding::DecodeError> {
+        let mut bytes = [0u8; 32];
+
+        d.read_exact(&mut bytes)
```

Comment:
Why use a different function for decoding than encoding? You could use `<[u8; 32] as Decodable>::consensus_decode(…)` instead (or a variation thereof).

The problem here is that you make assumptions about the inner workings of `<[u8; 32] as Encodable>::consensus_encode` by re-implementing the corresponding decoding functionality manually.

Reply:
Done